README.md

```diff
@@ -1,289 +1,610 @@
-
-Extra binding properties **SHOULD** also be defined by the bindable service, using one of the patterns defined in [Pointer to binding data](#pointer-to-binding-data).
-
-The data that is injected or mounted into the container may have a different name because of a few reasons:
-* the backing service may have chosen a different name (discouraged, but allowed).
-* a custom name may have been chosen using the `dataMappings` portion of the `ServiceBinding` CR.
-* a prefix may have been added to certain items in `ServiceBinding`, either globally or per service.  
-
-### Request service binding
-
-Binding is requested by the consuming application, or an entity on its behalf such as the [Runtime Component Operator](https://github.com/application-stacks/runtime-component-operator), via a custom resource that is applied in the same cluster where an implementation of this specification resides.
-
-Since the reference implementation for most of this specification is the [Service Binding Operator](https://github.com/redhat-developer/service-binding-operator) we will be using the `ServiceBinding` CRD, which resides in [this folder](https://github.com/redhat-developer/service-binding-operator/tree/master/deploy/crds), as the entity that holds the binding request.  
-
-**Note** - a few updates / enhancements are being proposed to the current `ServiceBinding` CR, tracked [here](https://github.com/application-stacks/service-binding-specification/issues/16#issuecomment-605309629).
-
-Sample CR
+## Example Directory Structure
+
+```plain
+$SERVICE_BINDING_ROOT
+├── account-database
+│   ├── type
+│   ├── connection-count
+│   ├── uri
+│   ├── username
+│   └── password
+└── transaction-event-stream
+    ├── type
+    ├── connection-count
+    ├── uri
+    ├── certificates
+    └── private-key
 ```
+
+# Service Binding
+
+A Service Binding describes the connection between a [Provisioned Service](#provisioned-service) and an [Application Projection](#application-projection).  It is codified as a concrete resource type.  Multiple Service Bindings can refer to the same service.  Multiple Service Bindings can refer to the same application.
+
+Restricting service binding to resources within the same namespace is strongly **RECOMMENDED**.  Cross-namespace service binding **SHOULD** be secured appropriately by the implementor to prevent attacks like privilege escalation and secret enumeration.
+
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference` to a `PodSpec`-able resource.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference` to a Provisioned Service-able resource.  A Service Binding resource **MAY** define a `.spec.name` which is the name of the service when projected into the application.
```

Comment:
`ObjectReference` is nice for 1:1 bindings, but aren't great for 1:N bindings.  In the knative bindings we also support specifying a `selector:` in place of `name:` to allow matching a class of resources (e.g. Jobs stamped out from a CronJob).

Reply:
I've loosened this up to be `ObjectReference`-able for now, and subsequent discussion/PR will address whether we should add the useful `selector:` behavior as well.

Reply:
We've talked about adding a label selector for applications as a spec extension

Reply:
if using a label to select multiple applications, I think we would be back with the issue of how to model the status for the application projections within a single `ServiceBinding` CR (same drawback for specifying multiple services)

Reply:
On the list for discussion. https://github.com/application-stacks/service-binding-specification/issues/51

Reply:
binding a single service into multiple applications is much easier than binding multiple services into a single application. (I don't want to think about many-to-many)

You're right that some of the status concerns resurface, but they are less dramatic. Since a label selector is scoped to a single kind, that kind will exist and be pod-specable, or not. A resource that doesn't exist is ok, as it won't match the label. The only real point of contention will be service name collisions.